chapter_04/p11_random_node.py

```diff
@@ -86,6 +89,43 @@ def get_random_node(self):
                 raise RuntimeError("Should not be possible")
 
 
+def _delete(node, key):
```

Comment:
It would feel nicer organizationally I think if we kept these functions as methods on the class but I don't feel strongly about it. You can leave them here if you think that's best.

Reply:
Good point, I couldn't code it that way since the node for delete changes during the recursion. Happy to learn from you as to how this can be done.

Reply:
yeah I'm not sure either